modules/social_features/social_group/social_group.api.php

```diff
@@ -169,6 +169,111 @@ function hook_social_group_group_visibility_description_alter($key, &$descriptio
   }
 }
 
+/**
+ * Alter the list of join plugin definitions.
+ *
+ * @param array $info
+ *   The join plugin definitions to be altered.
+ *
+ * @see \Drupal\social_group\Annotation\Join
+ * @see \Drupal\social_group\JoinManager
+ */
+function hook_social_group_join_info_alter(array &$info) {
+  if (isset($info['social_group_request_join'])) {
+    unset($info['social_group_request_join']['entityTypeId']);
+  }
+}
+
+/**
+ * Define join methods.
+ *
+ * @return array
+ *   An associative array of join method definitions. The keys are the
+ *   identifiers. The values are associative arrays that should contain the
+ *   following elements:
+ *   - title: The human-readable name of the join method. If this should be
+ *     translated, create a \Drupal\Core\StringTranslation\TranslatableMarkup
+ *     object.
+ *   - description: The description of the join method. The "@entity_type_id"
+ *     placeholder will be replaced automatically. If this should be translated,
+ *     create a \Drupal\Core\StringTranslation\TranslatableMarkup object.
+ *   - icon: The join method unique icon.
+ *   - weight: Integer weight used for sorting join methods.
+ *
+ * @see _social_group_allowed_values_callback()
+ * @see social_group_allowed_join_method_description()
+ * @see social_group_form_alter()
+ *
+ * @ingroup social_group_api
+ */
+function hook_social_group_join_method_info() {
+  return [
+    'direct' => [
+      'title' => t('Open to join'),
+      'description' => t('users can join this @entity_type_id without approval.'),
+      'icon' => 'join_open',
+      'weight' => 10,
+    ],
+    'added' => [
+      'title' => t('Invite only'),
+      'description' => t('users can only join this @entity_type_id if they are
+added/invited by @entity_type_id managers.'),
+      'icon' => 'invite',
+      'weight' => 20,
+    ],
+  ];
+}
+
+/**
+ * Define entity type bundles that support join methods.
+ *
+ * @return array
+ *   An array of entity types/bundles definitions that support join methods. The
+ *   values are associative arrays that should contain the following elements:
+ *   - entity_type_id: The entity type ID.
+ *   - bundle: (optional) The bundle(s).
+ *   - field: (optional) The field contains a list of supported join methods
+ *     when the "method" item isn't defined. Otherwise, the field indicates if
+ *     an entity can use the join method defined in the "method" item.
+ *   - method: (optional) The join method(s).
+ *
+ * @see \Drupal\social_group\JoinManager::relations()
+ *
+ * @ingroup social_group_api
+ */
+function hook_social_group_join_method_usage() {
+  return [
+    [
+      'entity_type_id' => 'group',
```

Comment:
@OH17L I am thinking that we can make this property name a bit shorter and use just `entity_type`.
I see that we already use such property in similar cases. See [here](https://github.com/goalgorilla/open_social/search?p=3&q=entity_type).

Reply:
Done.